File: 1_dfs_bfs/이준우/네트워크.kt
class Solution {
    fun solution(n: Int, computers: Array<IntArray>): Int {
        val visited = BooleanArray(n)
        var answer = 0

        fun bfs(start: Int) {
            val q = ArrayDeque<Int>()
            q.add(start)
            visited[start] = true

            while(q.isNotEmpty()) {
                val cur = q.removeLast()

                for(i in 0..<n) {
                    // 연결이 있으면서, 방문하지 않은 경우
                    if(computers[cur][i] == 1 && !visited[i]) {
                        visited[i] = true
                        q.add(i)
                    }
                }
            }
        }

        repeat(n) {
            if(!visited[it]) {
                answer += 1
                bfs(it)
            }
        }

        return answer
    }
}

File: 1_dfs_bfs/이준우/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    def bfs(start):
        q = deque([start])
        visited[start] = True
        
        while q:
            cur = q.popleft()
            
            for i in range(n):
                if computers[cur][i] == 1 and not visited[i]:
                    q.append(i)
                    visited[i] = True
    
    
    for i in range(n):
        if not visited[i]:
            answer += 1
            bfs(i)
    
    return answer